solvers/modopt_fista.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+
+from benchopt import BaseSolver
+from benchopt import safe_import_context
+
+
+with safe_import_context() as import_ctx:
+    from modopt.opt.algorithms import ForwardBackward
+    from modopt.opt.proximity import SparseThreshold
+    from modopt.opt.linear import Identity
+    from modopt.opt.gradient import GradBasic
+
+
+class Solver(BaseSolver):
+    name = 'ModOpt-FISTA'
+    stop_strategy = 'iteration'
+    install_cmd = 'conda'
+    requirements = [
+        'pip:git+https://github.com/CEA-COSMIC/ModOpt.git',
+    ]
+    parameters = {
+        'restart_strategy': ['greedy', 'adaptive-1'],
+    }
+    references = [
+        'S. Farrens, A. Grigis, L. El Gueddari, Z. Ramzi, G. R. Chaithya, '
+        'S. Starck, B. Sarthou, H. Cherkaoui, P. Ciuciu and J.-L. Starck, '
+        '"PySAP: Python Sparse Data Analysis Package for multidisciplinary '
+        'image processing", Astronomy and Computing, vol. 32, '
+        ' pp. 100402 (2020)'
+    ]
+    support_sparse = False
+
+    def set_objective(self, X, y, lmbd, fit_intercept):
+        self.X, self.y, self.lmbd = X, y, lmbd
```

Comment:
To mitigate this issue, we could simply create a copy of `X, y`?

Reply:
yep that's a nice easy fix. But it may take a bit of time, and it takes twice the memory, so if they can fix the issue on their side, I think it's also nice. So overall I'll wait for feedback